mode incorrect on open file.

**What happened**:
```
root@bench-01:~/juicefs#cat  .github/scripts/hypo/acl_test.py 
    def test_abc(self):
        state = JuicefsMachine()
        v1 = state.init_folders()
        v2 = state.create_file(content=b'\x11', file_name='pssw', mode='x', parent=v1, umask=208, user='root')
        v5 = state.hardlink(dest_file=v2, link_file_name='qtgl', parent=v1, umask=360, user='root')
        v6 = state.hardlink(dest_file=v2, link_file_name='uixm', parent=v1, umask=430, user='root')
        v10 = state.create_file(content=b'rH\xb4\xe5', file_name='xptk', mode='x', parent=v1, umask=258, user='root')
        v11 = state.create_file(content=b'\x19C', file_name='nnxx', mode='x', parent=v1, umask=54, user='root')
        v17 = state.hardlink(dest_file=v6, link_file_name='fifn', parent=v1, umask=110, user='root')
        v19 = state.create_file(content=b'x-a\xfc', file_name='aekf', mode='w', parent=v1, umask=145, user='root')
        v22 = state.rename_file(entry=v5, new_entry_name='brct', parent=v1, umask=507, user='root')
        v23 = state.hardlink(dest_file=v6, link_file_name='fpkc', parent=v1, umask=60, user='root')
        v26 = state.create_file(content=b'\xa5\x9e', file_name='vpbg', mode='a', parent=v1, umask=214, user='root')
        v30 = state.hardlink(dest_file=v10, link_file_name='bvfd', parent=v1, umask=391, user='root')
        v32 = state.create_file(content=b'', file_name='vrim', mode='a', parent=v1, umask=444, user='root')
        v36 = state.create_file(content=b'', file_name='mexk', mode='x', parent=v1, umask=438, user='root')
        v37 = state.create_file(content=b'|\xc9nN\xb0\x16', file_name='pfjn', mode='a', parent=v1, umask=424, user='root')
        v39 = state.create_file(content=b'', file_name='ebtc', mode='x', parent=v1, umask=111, user='root')
        v41 = state.hardlink(dest_file=v39, link_file_name='rxlx', parent=v1, umask=112, user='root')
        v42 = state.rename_file(entry=v39, new_entry_name='scsg', parent=v1, umask=192, user='root')
        v45 = state.create_file(content=b'\t\xc3\x01W^&\x00n', file_name='kpyr', mode='x', parent=v1, umask=394, user='root')
        v47 = state.hardlink(dest_file=v42, link_file_name='myqu', parent=v1, umask=67, user='root')
        v54 = state.rename_file(entry=v30, new_entry_name='frbf', parent=v1, umask=133, user='root')
        v57 = state.hardlink(dest_file=v54, link_file_name='aibl', parent=v1, umask=256, user='root')
        state.set_acl(default=True, entry=v57, group='group4', group_perm=set(), logical=True, mask={'r', 'w', 'x'}, not_recalc_mask=True, other_perm={'r', 'w', 'x'}, physical=False, recalc_mask=False, recursive=False, set_mask=False, sudo_user='root', user='user2', user_perm=set())
        v62 = state.create_file(content=b'0\x8c', file_name='gxht', mode='a', parent=v1, umask=402, user='root')
        v72 = state.create_file(content=b'\x92K\xf6@\x04\x1c\xed\x047\xad\xc4\xba\\\xccI', file_name='stjt', mode='w', parent=v1, umask=115, user='root')
        v78 = state.set_acl(default=True, entry=v57, group='group2', group_perm={'r', 'w', 'x'}, logical=True, mask={'r', 'w', 'x'}, not_recalc_mask=False, other_perm={'r'}, physical=True, recalc_mask=False, recursive=True, set_mask=True, sudo_user='root', user='root', user_perm=set())
        v81 = state.set_acl(default=False, entry=v41, group=v1, group_perm=set(), logical=True, mask=set(), not_recalc_mask=True, other_perm=set(), physical=False, recalc_mask=True, recursive=False, set_mask=True, sudo_user='root', user='root', user_perm=set())
        v82 = state.hardlink(dest_file=v32, link_file_name='ahhw', parent=v1, umask=52, user='root')
        v84 = state.set_acl(default=False, entry=v1, group='root', group_perm=set(), logical=False, mask={'r', 'w', 'x'}, not_recalc_mask=True, other_perm=set(), physical=False, recalc_mask=False, recursive=False, set_mask=False, sudo_user='root', user='root', user_perm={'r'})
        v88 = state.create_file(content=b'\x15', file_name='fuod', mode='a', parent=v1, umask=132, user='root')
        v89 = state.create_file(content=b'\x03\x07\x01', file_name='aeza', mode='a', parent=v1, umask=21, user='root')
        v90 = state.set_acl(default=True, entry=v88, group='user2', group_perm=set(), logical=True, mask=set(), not_recalc_mask=False, other_perm={'r'}, physical=False, recalc_mask=True, recursive=True, set_mask=True, sudo_user='root', user='user2', user_perm=set())
        v98 = state.set_acl(default=False, entry=v1, group='root', group_perm={'r'}, logical=True, mask=set(), not_recalc_mask=True, other_perm={'w'}, physical=False, recalc_mask=True, recursive=False, set_mask=True, sudo_user='root', user='root', user_perm=set())
        v99 = state.hardlink(dest_file=v37, link_file_name='kyhx', parent=v1, umask=125, user='root')
        v121 = state.set_acl(default=False, entry=v1, group='user3', group_perm={'r'}, logical=True, mask={'x'}, not_recalc_mask=True, other_perm=set(), physical=False, recalc_mask=True, recursive=True, set_mask=True, sudo_user='root', user='root', user_perm={'r', 'w', 'x'})
        state.open(file=v37, flags=[2], mode=3423, umask=362, user='root')
        state.teardown()

root@bench-01:~/juicefs# python3  .github/scripts/hypo/acl_test.py -k test_abc
setup_logger ./log1
setup_logger ./log2
__init__
duration is 0.02684760093688965
2024-04-12 14:21:36,963 - INFO - do_create_file /tmp/fsrand/pssw with mode x succeed
2024-04-12 14:21:36,964 - INFO - do_create_file /tmp/jfs/fsrand/pssw with mode x succeed
2024-04-12 14:21:36,964 - INFO - do_hardlink /tmp/fsrand/pssw /tmp/fsrand/qtgl succeed
2024-04-12 14:21:36,965 - INFO - do_hardlink /tmp/jfs/fsrand/pssw /tmp/jfs/fsrand/qtgl succeed
2024-04-12 14:21:36,965 - INFO - do_hardlink /tmp/fsrand/pssw /tmp/fsrand/uixm succeed
2024-04-12 14:21:36,966 - INFO - do_hardlink /tmp/jfs/fsrand/pssw /tmp/jfs/fsrand/uixm succeed
2024-04-12 14:21:36,966 - INFO - do_create_file /tmp/fsrand/xptk with mode x succeed
2024-04-12 14:21:36,966 - INFO - do_create_file /tmp/jfs/fsrand/xptk with mode x succeed
2024-04-12 14:21:36,967 - INFO - do_create_file /tmp/fsrand/nnxx with mode x succeed
2024-04-12 14:21:36,967 - INFO - do_create_file /tmp/jfs/fsrand/nnxx with mode x succeed
2024-04-12 14:21:36,967 - INFO - do_hardlink /tmp/fsrand/uixm /tmp/fsrand/fifn succeed
2024-04-12 14:21:36,968 - INFO - do_hardlink /tmp/jfs/fsrand/uixm /tmp/jfs/fsrand/fifn succeed
2024-04-12 14:21:36,968 - INFO - do_create_file /tmp/fsrand/aekf with mode w succeed
2024-04-12 14:21:36,969 - INFO - do_create_file /tmp/jfs/fsrand/aekf with mode w succeed
2024-04-12 14:21:36,969 - INFO - do_rename /tmp/fsrand/qtgl /tmp/fsrand/brct succeed
2024-04-12 14:21:36,969 - INFO - do_rename /tmp/jfs/fsrand/qtgl /tmp/jfs/fsrand/brct succeed
2024-04-12 14:21:36,969 - INFO - do_hardlink /tmp/fsrand/uixm /tmp/fsrand/fpkc succeed
2024-04-12 14:21:36,970 - INFO - do_hardlink /tmp/jfs/fsrand/uixm /tmp/jfs/fsrand/fpkc succeed
2024-04-12 14:21:36,970 - INFO - do_create_file /tmp/fsrand/vpbg with mode a succeed
2024-04-12 14:21:36,971 - INFO - do_create_file /tmp/jfs/fsrand/vpbg with mode a succeed
2024-04-12 14:21:36,971 - INFO - do_hardlink /tmp/fsrand/xptk /tmp/fsrand/bvfd succeed
2024-04-12 14:21:36,971 - INFO - do_hardlink /tmp/jfs/fsrand/xptk /tmp/jfs/fsrand/bvfd succeed
2024-04-12 14:21:36,971 - INFO - do_create_file /tmp/fsrand/vrim with mode a succeed
2024-04-12 14:21:36,972 - INFO - do_create_file /tmp/jfs/fsrand/vrim with mode a succeed
2024-04-12 14:21:36,972 - INFO - do_create_file /tmp/fsrand/mexk with mode x succeed
2024-04-12 14:21:36,973 - INFO - do_create_file /tmp/jfs/fsrand/mexk with mode x succeed
2024-04-12 14:21:36,973 - INFO - do_create_file /tmp/fsrand/pfjn with mode a succeed
2024-04-12 14:21:36,974 - INFO - do_create_file /tmp/jfs/fsrand/pfjn with mode a succeed
2024-04-12 14:21:36,974 - INFO - do_create_file /tmp/fsrand/ebtc with mode x succeed
2024-04-12 14:21:36,975 - INFO - do_create_file /tmp/jfs/fsrand/ebtc with mode x succeed
2024-04-12 14:21:36,975 - INFO - do_hardlink /tmp/fsrand/ebtc /tmp/fsrand/rxlx succeed
2024-04-12 14:21:36,975 - INFO - do_hardlink /tmp/jfs/fsrand/ebtc /tmp/jfs/fsrand/rxlx succeed
2024-04-12 14:21:36,975 - INFO - do_rename /tmp/fsrand/ebtc /tmp/fsrand/scsg succeed
2024-04-12 14:21:36,976 - INFO - do_rename /tmp/jfs/fsrand/ebtc /tmp/jfs/fsrand/scsg succeed
2024-04-12 14:21:36,976 - INFO - do_create_file /tmp/fsrand/kpyr with mode x succeed
2024-04-12 14:21:36,977 - INFO - do_create_file /tmp/jfs/fsrand/kpyr with mode x succeed
2024-04-12 14:21:36,977 - INFO - do_hardlink /tmp/fsrand/scsg /tmp/fsrand/myqu succeed
2024-04-12 14:21:36,977 - INFO - do_hardlink /tmp/jfs/fsrand/scsg /tmp/jfs/fsrand/myqu succeed
2024-04-12 14:21:36,977 - INFO - do_rename /tmp/fsrand/bvfd /tmp/fsrand/frbf succeed
2024-04-12 14:21:36,978 - INFO - do_rename /tmp/jfs/fsrand/bvfd /tmp/jfs/fsrand/frbf succeed
2024-04-12 14:21:36,978 - INFO - do_hardlink /tmp/fsrand/frbf /tmp/fsrand/aibl succeed
2024-04-12 14:21:36,978 - INFO - do_hardlink /tmp/jfs/fsrand/frbf /tmp/jfs/fsrand/aibl succeed
2024-04-12 14:21:36,978 - INFO - run_cmd: sudo -u root setfacl -d   --no-mask   -m u:user2:-,g:group4:-,o::rxw /tmp/fsrand/aibl
2024-04-12 14:21:36,983 - INFO - do_set_acl /tmp/fsrand/aibl {'user_perm': '-', 'group_perm': '-', 'other_perm': 'rxw'} failed:
setfacl: /tmp/fsrand/aibl: Only directories can have default ACLs
2024-04-12 14:21:36,983 - INFO - run_cmd: sudo -u root setfacl -d   --no-mask   -m u:user2:-,g:group4:-,o::rxw /tmp/jfs/fsrand/aibl
2024-04-12 14:21:36,987 - INFO - do_set_acl /tmp/jfs/fsrand/aibl {'user_perm': '-', 'group_perm': '-', 'other_perm': 'rxw'} failed:
setfacl: /tmp/jfs/fsrand/aibl: Only directories can have default ACLs
2024-04-12 14:21:36,988 - INFO - do_create_file /tmp/fsrand/gxht with mode a succeed
2024-04-12 14:21:36,988 - INFO - do_create_file /tmp/jfs/fsrand/gxht with mode a succeed
2024-04-12 14:21:36,989 - INFO - do_create_file /tmp/fsrand/stjt with mode w succeed
2024-04-12 14:21:36,989 - INFO - do_create_file /tmp/jfs/fsrand/stjt with mode w succeed
2024-04-12 14:21:36,989 - INFO - run_cmd: sudo -u root setfacl -d -R   -L -P -m u:root:-,g:group2:rxw,o::r,m::rxw /tmp/fsrand/aibl
run_cmd:getfacl /tmp/fsrand/aibl
2024-04-12 14:21:36,995 - INFO - do_set_acl /tmp/fsrand/aibl with u:root:-,g:group2:rxw,o::r,m::rxw succeed
2024-04-12 14:21:36,995 - INFO - run_cmd: sudo -u root setfacl -d -R   -L -P -m u:root:-,g:group2:rxw,o::r,m::rxw /tmp/jfs/fsrand/aibl
run_cmd:getfacl /tmp/jfs/fsrand/aibl
2024-04-12 14:21:37,000 - INFO - do_set_acl /tmp/jfs/fsrand/aibl with u:root:-,g:group2:rxw,o::r,m::rxw succeed
2024-04-12 14:21:37,000 - INFO - run_cmd: sudo -u root setfacl   --mask --no-mask   -m u:root:-,g::-,o::-,m::- /tmp/fsrand/rxlx
run_cmd:getfacl /tmp/fsrand/rxlx
2024-04-12 14:21:37,005 - INFO - do_set_acl /tmp/fsrand/rxlx with u:root:-,g::-,o::-,m::- succeed
2024-04-12 14:21:37,005 - INFO - run_cmd: sudo -u root setfacl   --mask --no-mask   -m u:root:-,g::-,o::-,m::- /tmp/jfs/fsrand/rxlx
run_cmd:getfacl /tmp/jfs/fsrand/rxlx
2024-04-12 14:21:37,011 - INFO - do_set_acl /tmp/jfs/fsrand/rxlx with u:root:-,g::-,o::-,m::- succeed
2024-04-12 14:21:37,011 - INFO - do_hardlink /tmp/fsrand/vrim /tmp/fsrand/ahhw succeed
2024-04-12 14:21:37,012 - INFO - do_hardlink /tmp/jfs/fsrand/vrim /tmp/jfs/fsrand/ahhw succeed
2024-04-12 14:21:37,012 - INFO - run_cmd: sudo -u root setfacl    --no-mask   -m u:root:r,g:root:-,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-04-12 14:21:37,017 - INFO - do_set_acl /tmp/fsrand/ with u:root:r,g:root:-,o::- succeed
2024-04-12 14:21:37,017 - INFO - run_cmd: sudo -u root setfacl    --no-mask   -m u:root:r,g:root:-,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-04-12 14:21:37,023 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:r,g:root:-,o::- succeed
2024-04-12 14:21:37,023 - INFO - do_create_file /tmp/fsrand/fuod with mode a succeed
2024-04-12 14:21:37,024 - INFO - do_create_file /tmp/jfs/fsrand/fuod with mode a succeed
2024-04-12 14:21:37,024 - INFO - do_create_file /tmp/fsrand/aeza with mode a succeed
2024-04-12 14:21:37,025 - INFO - do_create_file /tmp/jfs/fsrand/aeza with mode a succeed
2024-04-12 14:21:37,025 - INFO - run_cmd: sudo -u root setfacl -d -R --mask  -L  -m u:user2:-,g:user2:-,o::r,m::- /tmp/fsrand/fuod
run_cmd:getfacl /tmp/fsrand/fuod
2024-04-12 14:21:37,030 - INFO - do_set_acl /tmp/fsrand/fuod with u:user2:-,g:user2:-,o::r,m::- succeed
2024-04-12 14:21:37,030 - INFO - run_cmd: sudo -u root setfacl -d -R --mask  -L  -m u:user2:-,g:user2:-,o::r,m::- /tmp/jfs/fsrand/fuod
run_cmd:getfacl /tmp/jfs/fsrand/fuod
2024-04-12 14:21:37,036 - INFO - do_set_acl /tmp/jfs/fsrand/fuod with u:user2:-,g:user2:-,o::r,m::- succeed
2024-04-12 14:21:37,036 - INFO - run_cmd: sudo -u root setfacl   --mask --no-mask   -m u:root:-,g:root:r,o::w,m::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-04-12 14:21:37,041 - INFO - do_set_acl /tmp/fsrand/ with u:root:-,g:root:r,o::w,m::- succeed
2024-04-12 14:21:37,041 - INFO - run_cmd: sudo -u root setfacl   --mask --no-mask   -m u:root:-,g:root:r,o::w,m::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-04-12 14:21:37,046 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:-,g:root:r,o::w,m::- succeed
2024-04-12 14:21:37,047 - INFO - do_hardlink /tmp/fsrand/pfjn /tmp/fsrand/kyhx succeed
2024-04-12 14:21:37,047 - INFO - do_hardlink /tmp/jfs/fsrand/pfjn /tmp/jfs/fsrand/kyhx succeed
2024-04-12 14:21:37,047 - INFO - run_cmd: sudo -u root setfacl  -R --mask --no-mask -L  -m u:root:rxw,g:user3:r,o::-,m::x /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-04-12 14:21:37,052 - INFO - do_set_acl /tmp/fsrand/ with u:root:rxw,g:user3:r,o::-,m::x succeed
2024-04-12 14:21:37,053 - INFO - run_cmd: sudo -u root setfacl  -R --mask --no-mask -L  -m u:root:rxw,g:user3:r,o::-,m::x /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-04-12 14:21:37,062 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:rxw,g:user3:r,o::-,m::x succeed
2024-04-12 14:21:37,062 - INFO - do_open /tmp/fsrand/pfjn [2] 3423 succeed
2024-04-12 14:21:37,062 - INFO - do_open /tmp/jfs/fsrand/pfjn [2] 3423 succeed
F
======================================================================
FAIL: test_abc (__main__.TestFsrand2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".github/scripts/hypo/acl_test.py", line 163, in test_abc
    state.open(file=v37, flags=[2], mode=3423, umask=362, user='root')
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 105, in open
    flags = st_open_flags,
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 681, in rule_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 105, in open
    flags = st_open_flags,
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 802, in precondition_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 115, in open
    assert self.equal(result1, result2), f'\033[31mopen:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
AssertionError: open:
result1 is (0, 0, 18, '0o100010', 2)
result2 is (0, 0, 18, '0o100026', 2)

----------------------------------------------------------------------
Ran 1 test in 0.121s

FAILED (failures=1)

```




**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/vfs/vfs.go
+++ b/pkg/vfs/vfs.go
@@ -1059,6 +1059,7 @@ func (v *VFS) SetXattr(ctx Context, ino Ino, name string, value []byte, flags ui
 			return
 		}
 		err = v.Meta.SetFacl(ctx, ino, aclType, rule)
+		v.invalidateAttr(ino)
 	} else {
 		err = v.Meta.SetXattr(ctx, ino, name, value, flags)
 	}

